js/PulsarFormComponent.js

```diff
@@ -86,31 +89,56 @@ PulsarFormComponent.prototype.initDatePickers = function () {
     const datepickers = this.$html.find('[data-datepicker="true"]');
     let defaultDateFormat = 'DD/MM/YYYY';
 
+    moment.locale(window.navigator.userLanguage || window.navigator.language);
```

Comment:
I'm not sure we'll want to automatically change the locale based on user language, we'll quite likely need this to be an explicit user preference, or system preference, in a higher level UI.

Reply:
This is based on the browser's language setting. The same one which decides the `Accept-Language` request header. I'd assume this setting is set based on the system's language setting when the browser is installed.

Reply:
Sorry by 'system' I meant our software. The software should, by default, provide a consistent date format for all users which may then have the ability to be changed as a user preference, I just wanted to check that we're not inheriting the default locale based on the users location or OS preferences, it needs to be initially defined by the software constants.

Public XFP forms are a slight outlier in this situation where it's more appropriate for dates to be in the user's expected locale, but this needs to not be the default for the rest of the software, where more consistency is required.

Reply:
Interesting. I'd argue that the app should adapt to the user's setting where possible. That's based on my own opinion and the whole idea of "progressive" web apps. Would be interesting to get some research on the topic?

Reply:
This is based on the context of our software (the traditional use of 'people working at an organisation' rather than members of the public, which as I mentioned, is a non-default use case here). Dates need to be consistent across the software for members of the same organisation, for a single basic example: consider a call centre agent talking to a third-party service provider, talking about the date where an action should be taking place, it would be easy for confusion to happen if they're quoting dates to each other but unaware their preferences differ, and DD/MM vs MM/DD is being compared.

There's the ability/need for timezone/locale information to be used when displaying dates also, though that's outside the scope of this PR.

Reply:
Thinking outside the box, perhaps the actual solution to that problem would be to not display dates in that way. For example, always use the written month instead of the number - `12 June` over `12/06`. That way it's obvious no matter what. Say, for example, I live in the UK, and I move to the US to work - I would 100% get confused even when trying my best to stick to the correct date. If we always display in the word then this is avoided.

Reply:
Sure, however date formats are also a customer preference that we'd need to work with.

Reply:
Back on topic, however: given the reason I opened this PR is due to the differences server-side and client-side of rendering and evaluation of date-related things; it would help if this setting was able to accept an actual locale (such as `en_au`). That would mean I don't have to do any weird processing of the locale in the application in order to use it in Pulsar.

Reply:
I've reworked this to accept a `locale` option instead 👍